### DISCUSSION TITLE
Scaling when doing adjoint_nufft followed by nufft

### DISCUSSION THREAD

**Discussion by wouterzwerink** [mmuckley/torchkbnufft#69] [Q&A] [answered]
When data is put through the adjoint nufft and then the nufft, with some trajectory (radial, in this case), the resulting output is scaled by some factor. This is not unexpected, as the same happens in PyTorch when using fft and ifft sequentially. However, I am having trouble figuring out how to compute the scaling factor for torchkbnufft. Any ideas?

**Comment by mmuckley** [comment, marked as answer]
Hello @wouterzwerink, there are two factors: the FFT scaling and the scaling of density compensation.

The FFT scaling can be exactly compensated for by using the `"ortho"` option.

The density compensation, depending on the trajectory, cannot be perfectly compensated for. We have some algorithms in the package for calculating density compensation scaling based on papers in the literature, see [here](https://torchkbnufft.readthedocs.io/en/stable/generated/torchkbnufft.calc_density_compensation_function.html) for example.
